### ISSUE TITLE
[BUG] Llama 3.3 models not working

### ISSUE THREAD

**Issue by wypiki** [theroyallab/tabbyAPI#260]
### OS

Linux and Windows

### GPU Library

CUDA 12.x

### Python version

3.10

### Describe the bug

Loading any of the recently quantized LLama 3.3 70B-variants (bartowski, LoneStriker, Dracones), I get an exception.
This happens on Windows and Linux, and with all recent changes of TabbyAPI pulled and all dependencies updated.

### Reproduction steps

Setting a LLama 3.3-model in the config.yml and starting TabbyAPI.

### Expected behavior

Should load just like 3.1 or 3.1-Nemotron.

### Logs

```
INFO:     Using template "from_tokenizer_config" for chat completions.

Traceback (most recent call last):
  File "/home/ai_serviceuser/tabbyapi/tabbyAPI/start.py", line 295, in <module>
    entrypoint(converted_args)
  File "/home/ai_serviceuser/tabbyapi/tabbyAPI/main.py", line 164, in entrypoint
    asyncio.run(entrypoint_async())
  File "/home/ai_serviceuser/miniconda3/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/home/ai_serviceuser/miniconda3/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/home/ai_serviceuser/tabbyapi/tabbyAPI/main.py", line 70, in entrypoint_async
    await model.load_model(
  File "/home/ai_serviceuser/tabbyapi/tabbyAPI/common/model.py", line 112, in load_model
    async for _ in load_model_gen(model_path, **kwargs):
  File "/home/ai_serviceuser/tabbyapi/tabbyAPI/common/model.py", line 90, in load_model_gen
    async for module, modules in load_status:
  File "/home/ai_serviceuser/tabbyapi/tabbyAPI/backends/exllamav2/model.py", line 562, in load_gen
    async for value in iterate_in_threadpool(model_load_generator):
  File "/home/ai_serviceuser/tabbyapi/tabbyAPI/common/concurrency.py", line 30, in iterate_in_threadpool
    yield await asyncio.to_thread(gen_next, generator)
  File "/home/ai_serviceuser/miniconda3/lib/python3.10/asyncio/threads.py", line 25, in to_thread
    return await loop.run_in_executor(None, func_call)
  File "/home/ai_serviceuser/miniconda3/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/ai_serviceuser/tabbyapi/tabbyAPI/common/concurrency.py", line 20, in gen_next
    return next(generator)
  File "/home/ai_serviceuser/miniconda3/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 36, in generator_context
    response = gen.send(None)
  File "/home/ai_serviceuser/tabbyapi/tabbyAPI/backends/exllamav2/model.py", line 604, in load_model_sync
    self.tokenizer = ExLlamaV2Tokenizer(self.config)
  File "/home/ai_serviceuser/miniconda3/lib/python3.10/site-packages/exllamav2/tokenizer/tokenizer.py", line 130, in __init__
    self.tokenizer_model = ExLlamaV2TokenizerHF(path_hf)
  File "/home/ai_serviceuser/miniconda3/lib/python3.10/site-packages/exllamav2/tokenizer/hf.py", line 22, in __init__
    self.hf_tokenizer = Tokenizer.from_file(tokenizer_json)
Exception: data did not match any variant of untagged enum ModelWrapper at line 1251003 column 3

```

### Acknowledgements

- [X] I have looked for similar issues before submitting this one.
- [X] I have read the disclaimer, and this issue is related to a code bug. If I have a question, I will use the Discord server.
- [X] I understand that the developers have lives and my issue will be answered when possible.
- [X] I understand the developers of this program are human, and I will ask my questions politely.

**Comment by DocShotgun** [comment]
Odd... I was able to use a self-quant of Llama 3.3 70B Instruct today without any errors. You could try updating your exl2 to 0.2.6 (which isn't pinned yet in tabby), but arch-wise the model should be the same as 3.1.

**Comment by turboderp** [comment]
Could you examine the `tokenizer.json` file and make sure it was correctly downloaded? The error indicates that the file is corrupt somehow.

**Comment by turboderp** [comment]
Could you try either upgrading or downgrading your `tokenizers` package?

**Comment by wypiki** [comment]
tokenizers has version 0.21.0 as of now. 

Running
`pip install --force-reinstall -v "tokenizers==0.20.0"`

I get:
```
argilla 1.8.0 requires numpy<1.24.0, but you have numpy 1.26.1 which is incompatible.
argilla 1.8.0 requires pandas<2.0.0,>=1.0.0, but you have pandas 2.1.2 which is incompatible.
datasets 2.14.6 requires fsspec[http]<=2023.10.0,>=2023.1.0, but you have fsspec 2024.10.0 which is incompatible.
langchain 0.0.332 requires anyio<4.0, but you have anyio 4.0.0 which is incompatible.
torchaudio 2.1.0+cu118 requires torch==2.1.0+cu118, but you have torch 2.4.0 which is incompatible.
torchvision 0.16.0 requires torch==2.1.0, but you have torch 2.4.0 which is incompatible.
transformers 4.47.0 requires tokenizers<0.22,>=0.21, but you have tokenizers 0.20.0 which is incompatible.
```

**Comment by wypiki** [comment]
It reinstalled it anyway but I get the same error.

**Comment by turboderp** [comment]
Googling the specific error, `Exception: data did not match any variant of untagged enum ModelWrapper at line 1251003 column 3` it's definitely a known issue and seems to be specific to some versions of the `tokenizers` library (or to some versions of `transformers`). Perhaps try 0.19.1?

**Comment by DocShotgun** [comment]
I did a `pip list` in my venv and I'm using `tokenizers` 0.20.1, without `transformers` installed at all.

I'm noticing a lot of warnings in your log about mismatching dependency versions and packages that aren't part of tabby or exl2's dependencies at all. I'd suggest trying a fresh venv for tabby/exl2 by itself and see if that resolves it.

**Comment by Nodice** [comment]
I had the same issue, this fixed it for me and Llama-3.3-70B-Instruct-6.0bpw-h6-exl2 Loaded.

`pip install --force-reinstall -v "tokenizers==0.20.0"`

**Comment by bdashore3** [comment]
This seems like the issue. Added a min requirement to tokenizers, so an update will forcibly install it. Python's terrible packaging system strikes again.

See: bc3c154c9658fc57a8abdf8a097b53826d0c8400

I'll close this issue for the time being since it mainly seems solved.
